File: digital-banking-login/src/main/java/com/jfb/digital_banking_login/application/ports/in/LoginUseCase.java
package com.jfb.digital_banking_login.application.ports.in;

import com.jfb.digital_banking_login.adapters.controllers.request.LoginRequest;
import com.jfb.digital_banking_login.adapters.controllers.response.LoginResponse;

public interface LoginUseCase {
    LoginResponse execute(LoginRequest loginRequest);
}

File: digital-banking-fontend/app/src/main/java/com/jfb/digitalbanking/domain/usecase/LoginUseCase.kt
package com.jfb.digitalbanking.domain.usecase

import com.jfb.digitalbanking.data.model.LoginRequest
import com.jfb.digitalbanking.data.model.LoginResponse
import com.jfb.digitalbanking.data.repository.UserRepository

class LoginUseCase(private val userRepository: UserRepository) {

    suspend fun execute(loginRequest: LoginRequest): Result<LoginResponse> {
        return userRepository.login(loginRequest)
    }
}